lib/asciidoctor/substitutors.rb

```diff
@@ -322,9 +322,6 @@ def restore_passthroughs text, outer = true
       end
       subbed_text.include?(PASS_START) ? restore_passthroughs(subbed_text, false) : subbed_text
     end
-  ensure
-    # free memory if in outer call...we don't need these anymore
-    passes.clear if outer
```

Comment:
Since the comment say that we don't need to clear `passes` anymore I've removed it to fix this issue.

Reply:
I did a lot more digging into this issue to really understand it. While this seems like the straightforward fix, the problem is it allows the passthroughs Hash continue to grow throughout the lifetime of the document (each time substitutions are applied). This isn't significant for a one-time conversion, but it does become a problem if the document is being reused.

The problem we're observing stems from the nested call to apply_subs. It doesn't get the memo that it's nested. After trying out several approaches, I'm going to propose moving the outer/nested flag to apply_subs where we can have greater control over tracking where we are.